ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/ConversationIndexMemory.java

```diff
@@ -0,0 +1,255 @@
+                return;
+            }
+
+            if (map.containsKey("memory_index_name")) {
+                memoryMetaIndexName = (String) map.get("memory_index_name");
+            }
+            if (map.containsKey("memory_message_index_name")) {
+                memoryMessageIndexName = (String) map.get("memory_message_index_name");
+            }
+            if (map.containsKey(SESSION_ID)) {
+                String conversationId = (String) map.get(SESSION_ID);
+                GetRequest getRequest = new GetRequest(memoryMetaIndexName).id(conversationId);
+                client.get(getRequest, ActionListener.wrap(r -> {
+                    listener
+                        .onResponse(
+                            new ConversationIndexMemory(
+                                client,
+                                mlIndicesHandler,
+                                memoryMetaIndexName,
+                                memoryMessageIndexName,
+                                r.getId(),
+                                null
+                            )
+                        );
+                }, e -> { listener.onFailure(new IllegalArgumentException("Can't find conversation " + conversationId)); }));
+            } else if (map.containsKey("question")) {
+                String question = (String) map.get("question");
+                mlIndicesHandler.initMemoryMetaIndex(ActionListener.wrap(created -> {
+                    if (created) {
+                        IndexRequest indexRequest = new IndexRequest(memoryMetaIndexName);
+                        indexRequest.source(ImmutableMap.of("name", question, "created_time", Instant.now().toEpochMilli()));
+                        client.index(indexRequest, ActionListener.wrap(r -> {
+                            listener
+                                .onResponse(
+                                    new ConversationIndexMemory(
+                                        client,
+                                        mlIndicesHandler,
+                                        memoryMetaIndexName,
+                                        memoryMessageIndexName,
+                                        r.getId(),
+                                        null
+                                    )
+                                );
+                        }, e -> { listener.onFailure(e); }));
+                    } else {
+                        listener.onFailure(new RuntimeException("Failed to create memory meta index"));
+                    }
+                }, e -> { listener.onFailure(new RuntimeException("Failed to create memory meta index")); }));
+            } else {
+                listener.onFailure(new IllegalArgumentException("Invalid input parameter. Must set conversation id or question"));
+            }
+        }
+
+        public void create(String name, String memoryId, String appType, ActionListener<ConversationIndexMemory> listener) {
```

Comment:
We have Override method `create` in line 171
> public void create(Map<String, Object> map, ActionListener<ConversationIndexMemory> listener)

Why create another `create` method in this line ?

Reply:
refactored